src/Http/GenericRouter.php

```diff
@@ -172,4 +160,68 @@ private function createResponse(Response|View $input): Response
 
         return $input;
     }
+
+    private function matchStaticRoute(Request $request): ?MatchedRoute
+    {
+        $staticRoute = $this->getRoutes()[$request->method->value][$request->getPath()] ?? null;
+
+        if ($staticRoute === null) {
+            return null;
+        }
+
+        // TODO: why do we need to check this?
+        if ($staticRoute->isDynamic) {
+            return null;
+        }
+
+        return new MatchedRoute($staticRoute, []);
+    }
+
+    private function matchDynamicRoute(Request $request): ?MatchedRoute
```

Comment:
I think we're at a point where a dedicated `RouteMatcher` class might be useful.